alphastats/llm/llm_functions.py

```diff
@@ -1,15 +1,116 @@
 """Module for defining assistant functions for the ChatGPT model."""
 
-from typing import Dict, List
+from typing import Dict, List, Union
 
 import pandas as pd
+import streamlit as st
 
 from alphastats.dataset.dataset import DataSet
+from alphastats.gui.utils.ui_helper import StateKeys
 from alphastats.llm.enrichment_analysis import get_enrichment_data
-from alphastats.llm.uniprot_utils import get_gene_function
+from alphastats.llm.uniprot_utils import (
+    format_uniprot_annotation,
+    get_annotations_for_feature,
+)
+
+
+def get_annotation_from_store_by_feature_list(
+    features: list, annotation_store: dict
+) -> Union[dict, str, None]:
+    """
+    Retrieve the annotation from the store based on the feature representation.
+    This function searches for a feature in the feature-to-representation map that matches the given feature representation.
+    If a matching feature is found in the annotation store, its corresponding annotation is returned.
+    Args:
+        feature_repr (str): The representation of the feature to search for.
+        feature_to_repr_map (dict): A dictionary mapping features to their representations.
+        annotation_store (dict): A dictionary storing annotations for features.
+    Returns:
+        Union[dict, str, None]: The annotation corresponding to the feature representation if found, otherwise None.
+    """
+    for feature in features:
```

Comment:
the other features that come after a match are ignored on purpose?

Reply:
yes, the assymption is that only one entry with the repr makes it to the store, and if they do they are most likely identical or the llm couldn't tell them apart either. This is one reason why I'd rather do this controlled once upfront for all regulated features, as I can then pass the unique feature id (new PR).

Reply:
I updates the doc string accordingly